### PR TITLE
Look for domain in alt names instead of common name

### PR DESCRIPTION
**Pull request by navzam**
Fixes issue #24 (see issue for details)

### CODE REVIEW

File: validate-cert.js [hunk 1]
```diff
@@ -6,10 +6,18 @@ module.exports = function validate(pem_cert) {
...
-      return 'subjectAltName Check Failed'
+    // check that cert has a Subject Alternative Names (SANs) section
+    var altNameExt = cert.getExtension("subjectAltName")
+    if (!altNameExt) {
+      return 'invalid certificate validity (subjectAltName extension not present)'
+    }
+
+    // check that the domain echo-api.amazon.com is present in SANs section
+    var domainExists = altNameExt.altNames.some(function(name) {
+      return name.value.indexOf('echo-api.amazon.com') >= 0
```
Comment (mreinstein) [inline validate-cert.js]: I think this is problematic. simply checking to see if the cert name contains this string somewhere in it means people could register their own domains and get around this. For example, here is a site signed with a pem cert that would pass this check, but is owned by me, not amazon:  https://echo-api.amazon.com.reinstein.me/

Comment (navzam) [inline validate-cert.js]: Ah, good point. Leftover from the previous check of `=== -1`. I'll change to an exact match.

Comment (mreinstein) [inline validate-cert.js]: > I'll change to an exact match.

Ok, just please be very careful, as this piece is delicate and a small mistake could result in this module no longer successfully passing Amazon's skill tests.

Comment (navzam) [inline validate-cert.js]: The docs say the domain must be "present in the `Subject Alternative Names` (SANs) section of the signing certificate". I assume present means an exact match, but it is a bit vague.

The cert I'm getting from Alexa has that exact string ('echo-api.amazon.com') in the SAN section. I'll update to an exact match if you think that's the way to go.

Comment (mreinstein) [inline validate-cert.js]: sounds reasonable. thanks for digging in!

Comment (navzam) [inline validate-cert.js]: Pushed

**Pull request merged**